File: DCAutoParts/left/src/main/java/com/android/launcher/service/task/ResetLaunchAfterDataTask.java
package com.android.launcher.service.task;

import android.content.Context;

import com.android.launcher.util.AppUtils;
import com.android.launcher.util.LogUtils;

import com.dc.auto.library.module.module_db.entity.CarTravelTable;
import com.dc.auto.library.module.module_db.repository.CarTravelRepository;

/**
 * 重置启动后数据任务
 * @date： 2024/1/11
 * @author: 78495
*/
public class ResetLaunchAfterDataTask extends TaskBase{

    public ResetLaunchAfterDataTask(Context context) {
        super(context);
    }

    @Override
    public void run() {
        try {
            CarTravelTable carTravelTable = CarTravelRepository.getInstance().getData(context, AppUtils.getDeviceId(context));
            if(carTravelTable!=null){
                carTravelTable.setCurrentQtrip(0.0f);
                CarTravelRepository.getInstance().updateData(context,carTravelTable);
            }
            LogUtils.printI(TAG, "run---");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: DCAutoParts/left/src/main/java/com/android/launcher/service/task/kotlin/ResetLaunchAfterDataTask.kt
package com.android.launcher.service.task.kotlin

import com.android.launcher.util.AppUtils
import com.dc.auto.library.module.module_db.repository.CarTravelRepository
import dc.library.auto.task.api.step.SimpleTaskStep
import dc.library.auto.task.core.ThreadType
import dc.library.auto.task.logger.TaskLogger
import dc.library.auto.util.ContextUtil

/**
 * 重置启动后数据任务
 */
class ResetLaunchAfterDataTask : SimpleTaskStep() {

    override fun doTask() {
        try {
            val context = ContextUtil.getApplicationContext()
            val carTravelTable = CarTravelRepository.getInstance().getData(context, AppUtils.getDeviceId(context))
            if (carTravelTable != null) {
                carTravelTable.currentQtrip = 0.0f
                CarTravelRepository.getInstance().updateData(context, carTravelTable)
            }
            TaskLogger.i("ResetLaunchAfterDataTask run---")
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun getName(): String {
        return "ResetLaunchAfterDataTask"
    }

    override fun getThreadType(): ThreadType {
        // 任务的优先级不高，使用异步子线程执行
        // 耗时任务，比如第三方依赖库的初始化、大数据的预加载、磁盘读写操作等
        return ThreadType.ASYNC
    }
}